controllers/clustertopology_controller_compute.go

```diff
@@ -140,9 +141,129 @@ func (r *ClusterTopologyReconciler) getClass(ctx context.Context, cluster *clust
 	return class, nil
 }
 
-// getTemplate gets the object referenced in ref.
+// getCurrentState gets information about the current state of a Cluster by inspecting the state of the InfrastructureCluster,
+// the ControlPlane, and the MachineDeployments associated with the cluster.
+func (r *ClusterTopologyReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster, class clusterv1.ClusterClass) (*clusterTopologyState, error) {
+	var clusterState clusterTopologyState
```

Comment:
```suggestion
	clusterState := clusterTopologyState{
		cluster: cluster,
	}
```
Imho this makes it very obvious that we just use the cluster we get handed over here (In case you use this variant you of course don't have to set it below)

Reply:
👍